junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedInvocationNameFormatter.java

```diff
@@ -246,24 +245,27 @@ public synchronized void append(ArgumentsContext context, StringBuffer result) {
 			this.messageFormat.format(makeReadable(context.consumedArguments), result, new FieldPosition(0));
 		}
 
-		private @Nullable Object[] makeReadable(@Nullable Object[] arguments) {
-			@Nullable
-			Format[] formats = messageFormat.getFormatsByArgumentIndex();
-			@Nullable
-			Object[] result = Arrays.copyOf(arguments, Math.min(arguments.length, formats.length), Object[].class);
-			for (int i = 0; i < result.length; i++) {
-				if (formats[i] == null) {
-					result[i] = truncateIfExceedsMaxLength(StringUtils.nullSafeToString(arguments[i]));
-				}
-			}
-			return result;
+		@Nullable
+		private List<Object> makeReadable(@Nullable Object[] arguments) {
+			return makeReadable(arguments, messageFormat.getFormatsByArgumentIndex());
 		}
 
-		private @Nullable String truncateIfExceedsMaxLength(@Nullable String argument) {
-			if (argument != null && argument.length() > this.argumentMaxLength) {
-				return argument.substring(0, this.argumentMaxLength - 1) + ELLIPSIS;
-			}
-			return argument;
+		@Nullable
+		private List<Object> makeReadable(@Nullable Object[] arguments, Format[] formats) {
+			return IntStream.range(0, Math.min(arguments.length, formats.length)).mapToObj(
+				i -> makeReadable(arguments, i, formats[i])).toList();
+		}
+
+		@Nullable
+		private Object makeReadable(@Nullable Object[] arguments, int index, @Nullable Format format) {
+			return format != null ? arguments == null ? "" : arguments[index]
```

Comment:
- https://youtrack.jetbrains.com/issue/IDEA-375682/Condition-arguments-null-is-always-false

false positive: 

<img width="910" height="260" alt="Image" src="https://github.com/user-attachments/assets/006cce71-58ea-4f7f-b9b7-45b2476177b6" />